topics-project/app/(main)/home/page.js

```diff
@@ -1,21 +1,60 @@
 "use client";
+import React, { useState, useEffect } from 'react';
 import { Food } from "./_components/food";
 import { CalendarCarousel } from "./_components/calendarcarousel";
 import { CalendarSearch } from "lucide-react";
-import { useFetchMetrics, useFetchUserInfo } from "./action";
+import { useFetchMetrics, useFetchUserInfo, useFetchNutritionByDate } from "./action";
 import Macros from "./_components/macros";
+import { createClient } from "@/utils/supabase/client";
+import { format } from 'date-fns';
+
 export default function HomePage() {
-  {
-    /* Issue with full sized screen, white space on both corners and bottom, whitespace always present after food card (needs addressing) */
-  }
-  {
-    /* take array / storage from action . js and map to the food component (adjust UI of food component as well) */
-  }
+  const supabase = createClient();
+
+  const [userID, setUserID] = useState('');
+  const [selectedDate, setSelectedDate] = useState(new Date());
+
+  useEffect(() => {
+    const getUserID = async () => {
+      const { data, error } = await supabase.auth.getUser();
+      if (error) {
+        console.error("Error fetching user:", error.message);
+      } else {
+        setUserID(data.user.id);
+      }
+    };
+    getUserID();
+  }, []);
+
+  const handleDateChange = (newDate) => {
+    setSelectedDate(newDate); 
+  };
 
-  const name = "Johan";
-  const userID = "65567f92-7a4e-4d12-b1dc-1c4e2dd7343f";
+  const carouseldate = format(selectedDate, 'MM-dd-yyyy');
+    
   const { calories, fat, carbs, protein, loading, error } = useFetchMetrics(userID);
-  const { proteinGoal, fatGoal, carbsGoal } = useFetchUserInfo(userID);
+  const { proteinGoal, fatGoal, carbsGoal, userName } = useFetchUserInfo(userID);
+
+  const NutritionList = () => {
+  const { nutritionEntries, loading, error } = useFetchNutritionByDate(userID, carouseldate);
+
+  if (loading) return <p>Loading...</p>;
```

Comment:
Do we want to use the spinner component here instead?

Reply:
Will be addressing this in the next PR, thanks for the suggestion!